spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/DataTables.groovy

```diff
@@ -524,6 +562,334 @@ a ; b ; c
     4 | 5 || 9
   }
 
+  def 'data tables with pipes can be combined'() {
```

Comment:
This was already possible in an arguably more compact form already, could you maybe provide a better example for why this would be needed?
![image](https://user-images.githubusercontent.com/1841944/116407961-9796b680-a832-11eb-9ba9-f48529c6fef2.png)

Reply:
- because it is much more readable
- because it is much more writable
- because it is much more convenient
- because users asked for it
- ...

For example compare
```groovy
    where:
        y  | z
        -1 | 5
        -7 | 8
        0  | 1
        1  | 0

    combine:
        w | x
        0 | 1
        0 | 2
        1 | 3
        1 | 4
```

with

```groovy
    where:
        [w, x, y, z] <<
            [
                [
                    [0, 1],
                    [0, 2],
                    [1, 3],
                    [1, 4]
                ],
                [
                    [-1, 5],
                    [-7, 8],
                    [ 0, 1],
                    [ 1, 0]
                ]
            ]
                .combinations()
                *.flatten()
```

In the same manner you could ask why we have data tables at all.
They are just syntactic sugar over one data pipe per column.
But they are nicer to write, nicer to read, and people like them. :-)

Reply:
Besides that for the manual way, people must be pretty Groovy savvy to write it and to understand what is happening, where a sugar DSL makes it easy for them.

Reply:
could you add a test that asserts that both behave in the same way - it's what we'd expect, right?

Reply:
The existing tests should already check this, by checking the result, just that the combinations are pre-calculated.